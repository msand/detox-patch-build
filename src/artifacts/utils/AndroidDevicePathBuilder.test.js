const AndroidDevicePathBuilder = require('./AndroidDevicePathBuilder');

describe(AndroidDevicePathBuilder, () => {
  it('should generate process id dependent paths with a counter', () => {
    const date = new Date(2015, 5, 15, 15, 2, 58);
    const builder = new AndroidDevicePathBuilder(date);

    expect(builder.buildTemporaryArtifactPath('.png')).toMatchSnapshot();
    expect(builder.buildTemporaryArtifactPath('.png')).toMatchSnapshot();
  });

  it('should generate current timestamp string by default', () => {
    const builder = new AndroidDevicePathBuilder();
    expect(builder.buildTemporaryArtifactPath('.log')).toMatch(/^\/sdcard\/\d+_\d+\.log$/);
  });
});

