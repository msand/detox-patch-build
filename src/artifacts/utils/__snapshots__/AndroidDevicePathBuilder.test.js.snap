// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AndroidDevicePathBuilder should generate process id dependent paths with a counter 1`] = `"/sdcard/152580_0.png"`;

exports[`AndroidDevicePathBuilder should generate process id dependent paths with a counter 2`] = `"/sdcard/152580_1.png"`;
